vpr/SRC/route/route_timing.c

```diff
@@ -17,14 +18,31 @@
 #include "net_delay.h"
 #include "stats.h"
 #include "ReadOptions.h"
-
-
+ 
 using namespace std;
 
+/***************** Iterative connection based rerouting **********************/
 
+// conceptually works like rr_node_to_pin[net_index][sink_node_index] to get the pin index for that net
+// each net maps SINK node index -> PIN index for net
+// only need to be built once at the start since the SINK nodes never change
+// the reverse lookup of net_rr_terminals
+// do not query this directly, as rr_node_to_pin[inet][inode] will create the entry if it does not exist
+// instead, use it through functions convert_sink_node_to_pins_of_net and put_sink_rt_nodes_to_pins_lookup
+static vector<unordered_map<int,int>> rr_node_to_pin;
+
+// a property of each net, but only valid after pruning the previous route tree
+static vector<int> remaining_targets;
+static vector<t_rt_node*> reached_sinks;	// used to populate rt_node_of_sink after building route tree from traceback

```

Comment:
May want to rename first variable to rr_node_sink_to_pin (as only sinks go in it).  Or at least mention that explicitly in the comment.
Add a comment on use and size -- [0..num_pins_on_net-1] I think -- of remaining targets and reached sinks.  Do you index reached sink with the net pin number (i.e. entry 0 is invalid)?